src/service_control_client_impl.cc

```diff
@@ -95,34 +126,58 @@ ServiceControlClientImpl::~ServiceControlClientImpl() {
   // may call the flush callback. But since flush callback is disconnected,
   // we are OK.
   check_aggregator_->SetFlushCallback(NULL);
+  quota_aggregator_->SetFlushCallback(NULL);
   report_aggregator_->SetFlushCallback(NULL);
 }
 
+void ServiceControlClientImpl::AllocateQuotaFlushCallback(
+    const AllocateQuotaRequest& quota_request) {
+  AllocateQuotaResponse* quota_response = new AllocateQuotaResponse;
+
+  quota_transport_(
+      quota_request, quota_response,
+      [this, quota_request, quota_response](Status status) {
```

Comment:
when quota_request passes to lambda capture,  it is copied.   can we pass raw pointer?